noxfile.py

```diff
@@ -152,6 +155,72 @@ def lint(session: nox.Session) -> None:
     mypy(session)
 
 
+# TODO: node support is not tested yet - it should work if you require('xmlhttprequest') before
+# loading pyodide, but there is currently no nice way to do this with pytest-pyodide
+# because you can't override the test runner properties -
+# https://github.com/pyodide/pytest-pyodide/issues/118
+@nox.session(python="3.11")
+@nox.parametrize("runner", ["chrome"])
+def emscripten(session: nox.Session, runner: str) -> None:
+    """Test on Emscripten with Pyodide & Chrome"""
+    session.install("build","pyodide_build")
+    # build wheel into dist folder
+    session.run("python", "-m", "build")
+    # make sure we have a dist dir for pyodide
+    dist_dir = None
+    if "PYODIDE_ROOT" in os.environ:
+        # we have a pyodide build tree checked out
+        # use the dist directory from that
+        dist_dir = Path(os.environ["PYODIDE_ROOT"])
+    else:
+        # we don't have a build tree, get one
+        # that matches the version of pyodide build
+        import pyodide_build  # type: ignore[import]
+
+        pyodide_version = pyodide_build.__version__
+        pyodide_artifacts_path = Path(session.cache_dir) / f"pyodide-{pyodide_version}"
+        if not pyodide_artifacts_path.exists():
+            print("Fetching pyodide build artifacts")
+            session.run(
+                "wget",
+                f"https://github.com/pyodide/pyodide/releases/download/{pyodide_version}/pyodide-{pyodide_version}.tar.bz2",
+                "-O",
+                f"{pyodide_artifacts_path}.tar.bz2",
+            )
+            pyodide_artifacts_path.mkdir(parents=True)
+            session.run(
+                "tar",
+                "-xjf",
+                f"{pyodide_artifacts_path}.tar.bz2",
+                "-C",
+                str(pyodide_artifacts_path),
+                "--strip-components",
+                "1",
+            )
+
+        dist_dir = pyodide_artifacts_path
+    assert dist_dir is not None
+    assert dist_dir.exists()
+    if runner == "chrome":
+        # install chrome webdriver and add it to path
+        from webdriver_manager.chrome import ChromeDriverManager  # type: ignore[import]
+
+        driver = ChromeDriverManager().install()
+        session.env["PATH"] = f"{Path(driver).parent}:{session.env['PATH']}"
+
+        tests_impl(
+            session,
+            "emscripten-test",
+            pytest_extra_args=[
+                "--rt",
+                "chrome-no-host",
+                "--dist-dir",
+                str(dist_dir),
+                "test",
+            ],
+        )
```

Comment:
We should fail if we receive an unknown runner (ie `else: raise ValueError(...)`).